src/AppInstallerRepositoryCore/RepositorySource.cpp

```diff
@@ -327,6 +341,11 @@ namespace AppInstaller::Repository
             {
                 return Microsoft::PredefinedInstalledSourceFactory::Create();
             }
+            else if (Utility::CaseInsensitiveEquals(Rest::RestSourceFactory::Type(), type)

```

Comment:
>Utility::CaseInsensitiveEquals(Rest::RestSourceFactory::Type(), type) [](start = 21, length = 69)

From the comment above "Should always come from code, so no need for case insensitivity"
Shall we just use '==' for source type comparisons? Here and line 230

Reply:
That comment is because those are specifically internal sources.  A REST source may be configured externally, so yes, we should do case insensitive.

---
In reply to: [594688379](https://github.com/microsoft/winget-cli/pull/765#discussion_r594688379) [](ancestors = 594688379)

Reply:
So, we need case insensitivity check because the Rest source type doesnt come from code. I also realized that I missed adding case-insensitivity check in the rest factory. I will add it there as well.